plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -723,6 +739,32 @@ public void deallocate(Network network, NicProfile nic, VirtualMachineProfile vm
             } else {
                 super.deallocate(network, nic, vm);
             }
+
+            if (virtualMachine.getType() == VirtualMachine.Type.DomainRouter) {
```

Comment:
Can you extract this blob to a method?

Reply:
I am pretty sure @rafaelweingartner means the entire if block and not the enum check, @fmaximus

Reply:
Spot on. I was referring to the IF body, and not to the IF condition. We really need shorter methods and more unit tests.